js/plugins/dotprompt/src/prompt.ts

```diff
@@ -188,25 +202,49 @@ export class Dotprompt<Variables = unknown> implements PromptMetadata {
       tools: (options.tools || []).concat(this.tools || []),
       streamingCallback: options.streamingCallback,
       returnToolRequests: options.returnToolRequests,
-    } as GenerateOptions<z.ZodTypeAny, O>;
+    } as GenerateOptions<O, CustomOptions>;
   }
 
-  render<O extends z.ZodTypeAny = z.ZodTypeAny>(
-    opt: PromptGenerateOptions<Variables>
-  ): GenerateOptions<z.ZodTypeAny, O> {
-    return this._generateOptions<O>(opt);
+  /**
+   * Renders the prompt template based on user input.
+   *
+   * @param opt Options for the prompt template, including user input variables and custom model configuration options.
+   * @returns a `GenerateOptions` object to be used with the `generate()` function.
```

Comment:
Hmm, is this true? It looks like both `render()` and `generate()` take in the same parameters.

Reply:
Oops. I mean `generate` from `@genkit-ai/ai` here. Not sure how to disambiguate.

Reply:
Done